src/shared/nested-dropdown-list/types.ts

```diff
@@ -0,0 +1,84 @@
+import { Option } from "../../input-nested-select";
+
+export type TruncateType = "middle" | "end";
+export type ItemsLoadStateType = "loading" | "fail" | "success";
+export type Mode = "default" | "expand" | "collapse";
+
+export interface ListItemRenderArgs {
+    selected: boolean;
+}
+
+export interface DropdownDisplayProps<V1, V2, V3> {
+    /** Control the dropdown list when it is first opened */
+    mode?: Mode;
+    /** Expand the recursively selected key in dropdown list when it is first opened */
+    selectedKeyPath: string[];
```

Comment:
putting there because this `DropdownDisplayProps` will be shared for `InputNestedSelectProps`.

Reply:
it can be redeclared in `InputNestedSelectProps` though

and also in consideration that this will potentially be reused for the nested multi select

```
NestedDropdownListProps {
  selectedKeyPaths: string[][];
}

InputNestedSelectProps {
  selectedKeyPath: string[];
}
```